### ISSUE TITLE
Rules and [EnableIf(..., HideIfDisabled=true)] do not have effect on properties using [EmbedProperties] attribute

### ISSUE THREAD

**Issue by ThomasRonnowJensen** [opentap/opentap#1078]
**Describe the bug**
1: Adding a **Rule** to properties using the [EmbedProperties] attribute do not have any effect in the GUI.
2: **HideIfDisabled=true** in [EnableIf(...)] attribute do not have effect in the GUI for properties using [EmbedProperties] attribute
**Update:** \
3: Using attributes [Browsable(true)] and [XmlIgnore] do not disable serialization (to testplan) for properties using [EmbedProperties] attribute

**Environment (please complete the following information):**

 - OS: Linux
 - OpenTAP Version 9.20.0
 - Tested on KS8500

**To Reproduce**

``` c-sharp
public class CoverageLevel
{
    [Display(Name: "Inactivity Timer", Group: "", Order: 1, Description: "Send RRC connection release after this time(in ms) of network inactivity.")]
    [Unit("ms")]
    public int Inactivity_Timer { get; set; } = 100;
}

public enum NumberOfCoverageLevels
{
    [Display("0")]
    _0 = 0,
    [Display("1")]
    _1 = 1,
    [Display("2")]
    _2 = 2,
    [Display("3")]
    _3 = 3
}

public class MyTestStep : TestStep
{
    [Display("Number of Coverage Levels", Groups: new string[] { "eNB Settings", "Coverage Levels" }, Description: "Select number of coverage levels in the cell", Order: 1)]
    public NumberOfCoverageLevels numberOfCoverageLevels
    {
        get
        {
            return (NumberOfCoverageLevels)(CoverageLevels?.Count ?? 0);
        }
        set
        {
            if (CoverageLevels == null)
            {
                CoverageLevels = new List<CoverageLevel>();
            }
            if ((int)value < CoverageLevels.Count) //Note value is 1 indexed in the GUI
            {
                if ((int)value < CoverageLevelChoice)
                {
                    CoverageLevelChoice = (int)value; //To prevent exception error
                }
                CoverageLevels.RemoveRange((int)value, CoverageLevels.Count - (int)value);
            }
            while ((int)value > CoverageLevels.Count)
            {
                CoverageLevels.Add(new CoverageLevel());
            }
            if (CoverageLevels.Count > 0 && CoverageLevelChoice < 1)
            {
                CoverageLevelChoice = 1; //Make sure we have a valid selection (if we go from none to at least one element) to avoid exception error
            }
        }
    }

    [Browsable(false)]
    public List<CoverageLevel> CoverageLevels { get; set; } = new List<CoverageLevel>
    {
        new CoverageLevel() //Add a single Coverage Level to the list (we need at least 1)
    };

    [Browsable(false)]
    ///Contains a list of numbers presented by GUI and used as index in the CoverageLevels list
    ///Note: We want UI to be 1 indexed so we start range from 1. Remember to subtract 1 when accessing the
    ///      CoverageLevels list in <SelectedCoverageLevel> and when setting default value for <CoverageLevelChoice>
    public List<int> AvailableCoverageLevels => Enumerable.Range(1, CoverageLevels?.Count ?? 0).ToList();  //Start from 1 due to 1 indexed in the UI

    [Browsable(false)]
    public bool CoverageLevelEnabled => AvailableCoverageLevels?.Count > 0;

    [EnabledIf(nameof(CoverageLevelEnabled), HideIfDisabled = true)]
    [AvailableValues(nameof(AvailableCoverageLevels))]
    [Display("Coverage Level Selected", Groups: new string[] { "eNB Settings", "Coverage Levels" }, Description: "Select coverage level in the cell to modify", Order: 3)]
    public int CoverageLevelChoice { get; set; } = 1; //Start from 1 due to 1 indexed in the GUI (if CoverageLevels.Count==0 this will throw exception - better to set it in the constructor if CoverageLevels.Count>0)

    [EnabledIf(nameof(CoverageLevelEnabled), HideIfDisabled = true)]
    [EmbedProperties]
    [Browsable(true)]
    [XmlIgnore]
    [Display("Chosen Coverage Level", Groups: new string[] { "eNB Settings", "Coverage Levels" }, Description: "Selected coverage level to modify", Order: 4)]
    public CoverageLevel SelectedCoverageLevel => (CoverageLevels?.Count < 1 ? null : CoverageLevels[CoverageLevelChoice - 1]); //-1 due to 1 indexed in the UI

    public MyTestStep()
    {
      Rules.Add(() => (SelectedCoverageLevel.Inactivity_Timer  >= 0),
          "SelectedCoverageLevel.Inactivity_Timer >= 0",
          nameof(CoverageLevel.Inactivity_Timer));
          //nameof(SelectedCoverageLevel.Inactivity_Timer)); //Also tried without qualifying
     }
}
```
**Actual behavior**

When selecting a value that do not comply with the rule we do not see a red mark next to the parameter in the GUI

1: Setting "Inactivity timer" to value below 0 in the GUI do not give a red mark in the GUI
2: Setting "Number of Coverage Levels" to 0 in the GUI - the "Chosen Coverage Level" property is still visible in the GUI but should be hidden when "Number of Coverage Levels" is set to 0
**Update:** \
3: The 'SelectedCoverageLevel' property is serialized (to TapPlan) even thou [Browsable(true)] and [XmlIgnore] attributes are used. This leads to errors (seen in log) during serialization/deserialization when 'SelectedCoverageLevel' is NULL (which it is when CoverageLevels.Count < 1)

**Expected behavior**

1: A red mark next to the parameter in the GUI when selecting a value that do not comply with the rule
2: "Chosen Coverage Level" to be hidden when setting "Number of Coverage Levels" to 0 in the GUI
**Update:** \
3: The 'SelectedCoverageLevel ' property is not serialized

**Additional context**

N/A

**Comment by alnlarsen** [comment]
This is related to #682 and #58

**Comment by Denrask** [comment]
Fixed by #1147
